File: backup_java/dialogs/DialogBestiary.java
package it.paranoidsquirrels.idleguildmaster.ui.dialogs;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ListAdapter;
import android.widget.RadioGroup;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.viewbinding.ViewBinding;
import it.paranoidsquirrels.idleguildmaster.MainActivity;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.UIUtils;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.databinding.DialogBestiaryBinding;
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Area;
import java.util.ArrayList;
import java.util.function.Predicate;

/* JADX INFO: loaded from: classes3.dex */
public class DialogBestiary extends CustomDialog {
    private BaseAdapter adapter;
    public DialogBestiaryBinding binding;

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding getBinding() {
        return this.binding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void setBinding(ViewBinding viewBinding) {
        this.binding = (DialogBestiaryBinding) viewBinding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected String getTitle() {
        return getString(R.string.drawer_bestiary_title);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding inflate(LayoutInflater layoutInflater, ViewGroup viewGroup, boolean z) {
        DialogBestiaryBinding dialogBestiaryBindingInflate = DialogBestiaryBinding.inflate(layoutInflater, viewGroup, z);
        this.binding = dialogBestiaryBindingInflate;
        return dialogBestiaryBindingInflate;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void initialize(Bundle bundle) {
        ArrayList arrayList = new ArrayList(Utils.compileDungeonList());
        arrayList.removeIf(new Predicate() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBestiary$$ExternalSyntheticLambda0
            @Override // java.util.function.Predicate
            public final boolean test(Object obj) {
                return DialogBestiary.lambda$initialize$0((Area) obj);
            }
        });
        ArrayList arrayList2 = new ArrayList(Utils.compileRaidList());
        arrayList2.removeIf(new Predicate() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBestiary$$ExternalSyntheticLambda1
            @Override // java.util.function.Predicate
            public final boolean test(Object obj) {
                return DialogBestiary.lambda$initialize$1((Area) obj);
            }
        });
        constraintHeight(this.binding.dungeonsBestiaryList, arrayList.size() > 3);
        constraintHeight(this.binding.raidsBestiaryList, arrayList2.size() > 3);
        this.binding.dungeonsBestiaryList.setAdapter((ListAdapter) UIUtils.getBestiaryListAdapter(arrayList));
        this.binding.raidsBestiaryList.setAdapter((ListAdapter) UIUtils.getBestiaryListAdapter(arrayList2));
        this.binding.radiobuttonDungeons.setChecked(true);
        toggle();
    }

    static /* synthetic */ boolean lambda$initialize$0(Area area) {
        return !area.isUnlocked();
    }

    static /* synthetic */ boolean lambda$initialize$1(Area area) {
        return !area.isUnlocked();
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void attachListeners() {
        this.binding.radioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBestiary$$ExternalSyntheticLambda2
            @Override // android.widget.RadioGroup.OnCheckedChangeListener
            public final void onCheckedChanged(RadioGroup radioGroup, int i) {
                DialogBestiary.this.m224xbcf0f162(radioGroup, i);
            }
        });
        this.binding.close.setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBestiary$$ExternalSyntheticLambda3
            @Override // android.view.View.OnClickListener
            public final void onClick(View view) {
                DialogBestiary.this.m225x3b51f541(view);
            }
        });
    }

    /* JADX INFO: renamed from: lambda$attachListeners$2$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogBestiary, reason: not valid java name */
    /* synthetic */ void m224xbcf0f162(RadioGroup radioGroup, int i) {
        toggle();
    }

    /* JADX INFO: renamed from: lambda$attachListeners$3$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogBestiary, reason: not valid java name */
    /* synthetic */ void m225x3b51f541(View view) {
        dismiss();
    }

    private void toggle() {
        boolean z = this.binding.radioGroup.getCheckedRadioButtonId() == R.id.radiobutton_dungeons;
        this.binding.dungeonsBestiaryList.setVisibility(z ? 0 : 8);
        this.binding.raidsBestiaryList.setVisibility(z ? 8 : 0);
    }

    private void constraintHeight(View view, boolean z) {
        ConstraintLayout.LayoutParams layoutParams = (ConstraintLayout.LayoutParams) view.getLayoutParams();
        layoutParams.constrainedHeight = z;
        view.setLayoutParams(layoutParams);
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStart() {
        super.onStart();
        MainActivity.shownDialogBestiary = this;
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStop() {
        MainActivity.shownDialogBestiary = null;
        super.onStop();
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/ui/dialogs/DialogBestiary.kt
package it.paranoidsquirrels.idleguildmaster.ui.dialogs

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.viewbinding.ViewBinding
import it.paranoidsquirrels.idleguildmaster.MainActivity
import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.UIUtils
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.databinding.DialogBestiaryBinding
import it.paranoidsquirrels.idleguildmaster.storage.data.places.Area

class DialogBestiary : CustomDialog() {
    private var adapter: BaseAdapter? = null
    @JvmField
    var binding: DialogBestiaryBinding? = null

    override fun getBinding(): ViewBinding = binding!!

    override fun setBinding(viewBinding: ViewBinding) {
        binding = viewBinding as DialogBestiaryBinding
    }

    override fun getTitle(): String = getString(R.string.drawer_bestiary_title)

    override fun inflate(inflater: LayoutInflater, container: ViewGroup?, attachToRoot: Boolean): ViewBinding {
        val b = DialogBestiaryBinding.inflate(inflater, container, attachToRoot)
        binding = b
        return b
    }

    override fun initialize(arguments: Bundle?) {
        val b = binding ?: return
        val dungeons = Utils.compileDungeonList().filter { it.isUnlocked }
        val raids = Utils.compileRaidList().filter { it.isUnlocked && it != MainActivity.data.guildRequest && it != MainActivity.data.guildSiege }
        val other = listOfNotNull(MainActivity.data.guildRequest, MainActivity.data.guildSiege).filter { it.isUnlocked }
        constraintHeight(b.dungeonsBestiaryList, dungeons.size > 3)
        constraintHeight(b.raidsBestiaryList, raids.size > 3)
        constraintHeight(b.otherBestiaryList, other.size > 3)
        b.dungeonsBestiaryList.adapter = UIUtils.getBestiaryListAdapter(dungeons)
        b.raidsBestiaryList.adapter = UIUtils.getBestiaryListAdapter(raids)
        b.otherBestiaryList.adapter = UIUtils.getBestiaryListAdapter(other)
        b.radiobuttonDungeons.isChecked = true
        toggle()
    }

    override fun attachListeners() {
        val b = binding ?: return
        b.radioGroup.setOnCheckedChangeListener { _, _ ->
            toggle()
        }
        b.close.setOnClickListener {
            dismiss()
        }
    }

    private fun toggle() {
        val b = binding ?: return
        when (b.radioGroup.checkedRadioButtonId) {
            R.id.radiobutton_dungeons -> {
                b.dungeonsBestiaryList.visibility = View.VISIBLE
                b.raidsBestiaryList.visibility = View.GONE
                b.otherBestiaryList.visibility = View.GONE
            }
            R.id.radiobutton_raids -> {
                b.dungeonsBestiaryList.visibility = View.GONE
                b.raidsBestiaryList.visibility = View.VISIBLE
                b.otherBestiaryList.visibility = View.GONE
            }
            R.id.radiobutton_other -> {
                b.dungeonsBestiaryList.visibility = View.GONE
                b.raidsBestiaryList.visibility = View.GONE
                b.otherBestiaryList.visibility = View.VISIBLE
            }
        }
    }

    private fun constraintHeight(view: View, z: Boolean) {
        val layoutParams = view.layoutParams as ConstraintLayout.LayoutParams
        layoutParams.constrainedHeight = z
        view.setLayoutParams(layoutParams)
    }
}
